src/reference/asciidoc/overview.adoc

```diff
@@ -273,6 +273,84 @@ If you need to send a messages during startup, implement `ApplicationListener` a
 Alternatively, implement `SmartLifecycle`, put your bean in a late phase, and send the messages from the `start()` method.
 
 
+[[shaded]]
+=== Considerations When using Packaged (e.g. Shaded) Jars
+
+Spring Integration bootstraps certain features using Spring Framework's `SpringFactories` mechanism to load several `IntegrationConfigurationInitializer` classes.
+This includes the `-core` jar as well as certain others such as `-http`, `-jmx`, etc.
+The information for this process is stored in a file `META-INF/spring.factories` in each jar.
+
+Some developers prefer to repackage their application and all dependencies into a single jar using well-known tools, such as the https://maven.apache.org/plugins/maven-shade-plugin/[Apache Maven Shade Plugin].
+
+By default, the shade plugin will not merge the `spring.factories` files when producing the shaded jar.
+
+In addition to `spring.factories`, there are other `META-INF` files (`spring.handlers`, `spring.schemas`) used for XML configuration.
+These also need to be merged.
+
+IMPORTANT: https://docs.spring.io/spring-boot/docs/1.5.10.RELEASE/reference/html/executable-jar.html[Spring Boot's executable jar mechanism] takes a different approach in that it nests the jars, thus retaining each `spring.factories` file on the class path.
```

Comment:
Does `/current` sym-link work for us here instead of an explicit Spring Boot version?